contents/blog/the-posthog-array-1-39-0.mdx

```diff
@@ -0,0 +1,133 @@
+- [New: Added features for client and server-side libraries](#new-added-features-for-client-and-server-side-libraries)
+- [New: Format your insights to show durations](#new-format-your-insights-to-show-durations)
+- [New: Big, bold number chart type](#new-big-bold-number-chart-type)
+- [Improved: Turbo properties](#improved-turbo-properties)
+- [Improved: History log now shows changes to insights](#improved-history-log-now-shows-changes-to-insights)
+- [Apps: Census integration](#apps-census-integration)
+
+### Beta: Improving query performance by combining persons and events
+
+We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and adding person ids and properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release. Here's how:
+
+- **PostHog Cloud:** No action is needed, as we're rolling this test out steadily behind a feature flag. If you're eager to be included from the start, let us know in [Slack](/slack) to see about joining our beta group.
+-  **Self-hosted users:** First, you need to [run async migration number 0006](/docs/runbook/async-migrations). Afterwards, change your `PERSON_ON_EVENTS_ENABLED` [instance setting](/docs/self-host/configure/instance-settings) to `true`. As with other instance settings, you'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.
+
+Why are we doing this? Firstly, it makes queries significantly faster because we no longer have to join tables to get a result; we can just look up everything in the events table instead. One query in our internal tests showed a 400x increase in speed...but this beta will help us understand the improvement in real-world conditions. 
+
+Secondly, feedback showed that users weren't able to create queries based on person properties at the time of an event. By putting person properties on events, this becomes the new default, while still enabling you to to filter insights based on the latest properties using cohorts. 
+
+There'll be some slight changes to how filtering by unique users or person properties work as a result of this change. For more information on this and to find out how we've implemented the change behind the scenes, check out our [overview of the ingestion pipeline](/docs/architecture/ingestion-pipeline).
+
+## New: Added features for client and server-side libraries
+
+While we continue to recommend using Javascript if possible, we've made significant updates to our other client libraries so that Android, iOS, Flutter and React Native now all support feature flags. 
+
+We've also updated our server-side libraries with local feature flag evaluation. This is huge because you can now use feature flags in your backend without worrying about latency. Evaluating locally reduces the computation time from an average of 400ms to 10ms!
```

Comment:
```suggestion
We've also updated our server-side libraries with local feature flag evaluation. This is huge because you can now use feature flags in your backend without worrying about latency. Evaluating locally reduces the computation time from an average of 400ms to 10ms!

Also, in case you missed it previously, we've rewritten the React Native library, bringing it up to a 2.0 which is [expo-compatible](https://expo.dev/). We actually did this a little while ago and didn't announce it at the time, which is why you may have missed it before. Sorry!
```

@mariusandra this cover it?

Reply:
It was released on 2nd August, so this array is the right one: https://github.com/PostHog/posthog-js-lite/pull/10

No need for the last line :)

Reply:
oh, maybe worth covering that the minimum PostHog version requirement for all these new libraries is 1.38 (we mention it in the CHANGELOG, but would be nice to highlight here as well for self hosted peeps planning to upgrade libraries after reading this)

Reply:
@benjackwhite is the right person to 👍 / 👎 this

Reply:
I forgot that he's off today. I think the RN blob requires its own bullet point / header, as these big things got released all at once:
- React Native integration fully rewritten in TS. This means no external iOS/Android library dependencies and thus it now works with Expo.
- Added support for magic autocapture (that's a new for the react native library - only custom instrumented events were supported before, now we do magic)
- Added support for feature flags and group analytics (most libraries got these recently)

Reply:
Added a commit to reflect this and will add to the email now too. Thanks!